src/System.IO.Pipes.AccessControl/tests/NamedPipeTests/NamedPipeTest.AclExtensions.cs

```diff
@@ -40,12 +40,27 @@ public void GetAccessControl_NamedPipe_BeforeWaitingToConnect()
             Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
+        public void SetAccessControl_NamedPipeClientStream_Broken()
+        {
+            NamedPipeClientStream client = GetAccessControl_NamedPipeClientStream_Broken();
+            Assert.Throws<InvalidOperationException>(() => client.SetAccessControl(new PipeSecurity()));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "On Desktop we throw an exception based on the internal PipeStream state. In .NET Core, we do not have access to that member since Pipes.AccessControl is its own assembly.")]
+        public void SetAccessControl_NamedPipeClientStream_Broken_Desktop()
+        {
+            NamedPipeClientStream client = GetAccessControl_NamedPipeClientStream_Broken();
+            Assert.Throws<IOException>(() => client.SetAccessControl(new PipeSecurity()));
```

Comment:
>You don't directly access the PipeStream state you just catch the exception that would come out and then rethrow the exception that gets thrown on .NET Framework.

There isn't an exception to catch though is the issue. It's the `SetAccessControl` method itself that is supposed to be doing validation on the state of the PipeStream. The actual handle pipesecurity call isn't going to throw an exception in the PipeStream is *broken* - only if it is closed.